client/state/analytics/actions/enhance-with-user-is-dev-account.js

```diff
@@ -0,0 +1,31 @@
+import { set } from 'lodash';
+import { ANALYTICS_EVENT_RECORD } from 'calypso/state/action-types';
+import getUserSetting from 'calypso/state/selectors/get-user-setting';
+
+/**
+ * Enhances any Redux action that denotes the recording of an analytics event with
+ * an additional property `user_is_dev_account` which specifies whether the user has
+ * the `is_dev_account` setting ("I am a developer") enabled.
+ *
+ * @param {Object} action - Redux action as a plain object
+ * @param {Function} getState - Redux function that can be used to retrieve the current state tree
+ * @returns {import('redux').AnyAction} the new Redux action
+ * @see client/state/utils/withEnhancers
+ */
+export function enhanceWithUserIsDevAccount( action, getState ) {
```

Comment:
The enhancer implementation is done referencing what we have for including the `site_type` property: https://github.com/Automattic/wp-calypso/blob/d8a7fa9286df26282d90552d9644eaed357186be/client/state/analytics/actions/enhance-with-site-type.js#L13